service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionTemplateDataGenerationService.java

```diff
@@ -0,0 +1,742 @@
+    private static final String NEW_LINE = "\n";
+    private static final String SPACE_DELIMITER = " ";
+    private static final String DEFAULT_STRING = "-";
+    private static final String CONFIDENTIAL = "Confidential";
+
+    private final Time time;
+    private final UserDetailsService userDetailsService;
+    private final HmctsCourtLookupConfiguration courtLookupConfiguration;
+
+    public DocmosisCaseSubmission getTemplateData(final CaseData caseData) {
+        DocmosisCaseSubmission.Builder applicationFormBuilder = DocmosisCaseSubmission.builder();
+
+        applicationFormBuilder
+            .applicantOrganisations(getApplicantsOrganisations(caseData.getAllApplicants()))
+            .respondentNames(getRespondentsNames(caseData.getAllRespondents()))
+            .courtName(courtLookupConfiguration.getCourt(caseData.getCaseLocalAuthority()).getName())
+            .submittedDate(formatDateDisplay(time.now().toLocalDate()))
+            .ordersNeeded(getOrdersNeeded(caseData.getOrders()))
+            .directionsNeeded(getDirectionsNeeded(caseData.getOrders()))
+            .allocation(caseData.getAllocationProposal())
+            .hearing(buildDocmosisHearing(caseData.getHearing()))
+            .hearingPreferences(buildDocmosisHearingPreferences(caseData.getHearingPreferences()))
+            .internationalElement(buildDocmosisInternationalElement(caseData.getInternationalElement()))
+            .risks(buildDocmosisRisks(caseData.getRisks()))
+            .factorsParenting(buildDocmosisFactorsParenting(caseData.getFactorsParenting()))
+            .respondents(buildDocmosisRespondents(caseData.getAllRespondents()))
+            .applicants(buildDocmosisApplicants(caseData.getAllApplicants(), caseData.getSolicitor()))
+            .children(buildDocmosisChildren(caseData.getAllChildren()))
+            .others(buildDocmosisOthers(caseData.getAllOthers()))
+            .proceeding(buildDocmosisProceedings(caseData.getAllProceedings()))
+            .relevantProceedings(getValidAnswerOrDefaultValue(caseData.getRelevantProceedings()))
+            .groundsForEPOReason(isNotEmpty(caseData.getOrders())
+                ? getGroundsForEPOReason(caseData.getOrders().getOrderType(),
+                    caseData.getGroundsForEPO())
+                : DEFAULT_STRING)
+            .groundsThresholdReason(buildGroundsThresholdReason(caseData.getGrounds()))
+            .thresholdDetails(getThresholdDetails(caseData.getGrounds()))
+            .annexDocuments(buildDocmosisAnnexDocuments(caseData))
+            .userFullName(userDetailsService.getUserName())
+            .courtseal(COURT_SEAL.getValue());
+
+        return applicationFormBuilder.build();
+    }
+
+    public DocmosisCaseSubmission populateDocmosisCaseSubmissionWithCaseNumber(
+        final DocmosisCaseSubmission submittedCase, final long caseNumber) {
+        return submittedCase.toBuilder().caseNumber(String.valueOf(caseNumber)).build();
```

Comment:
What if you just set the case number to not be final and then just set it, you then wouldn't need to make a new version of the object and this method then wouldn't need to return anything as well